src/java/org/apache/cassandra/tools/nodetool/ConsensusMigrationAdmin.java

```diff
@@ -0,0 +1,111 @@
+    {
+        @Option(title = "start_token", name = {"-st", "--start-token"}, description = "Use -st to specify a token at which the repair range starts")
+        private String startToken = null;
+
+        @Option(title = "end_token", name = {"-et", "--end-token"}, description = "Use -et to specify a token at which repair range ends")
+        private String endToken = null;
+
+        @Option(title = "target_protocol", name = {"-tp", "--target-protocol"}, description = "Use -tp to specify what consensus protocol should be migrated to", required=true)
+        private String targetProtocol = null;
+
+        @Arguments(usage = "[<keyspace> <tables>...]", description = "The keyspace followed by one or many tables")
+        private List<String> schemaArgs = new ArrayList<>();
+
+        protected void execute(NodeProbe probe)
+        {
+            checkNotNull(startToken, "Start token must be specified");
+            checkNotNull(endToken, "End token must be specified");
+            checkArgument(schemaArgs.size() >= 2, "Must specify a keyspace and at least one table");
+            checkArgument((endToken != null && startToken != null) || (endToken == null && startToken == null), "Must specify start and end token together");
+            String maybeRangesStr = startToken != null ? startToken + ":" + endToken : null;
+            List<String> keyspaceNames = schemaArgs.size() > 0 ? singletonList(schemaArgs.get(0)) : emptyList();
+            List<String> maybeTableNames = schemaArgs.size() > 1 ? schemaArgs.subList(1, schemaArgs.size()) : null;
+            probe.getStorageService().migrateConsensusProtocol(targetProtocol, keyspaceNames, maybeTableNames, maybeRangesStr);
+        }
+    }
+
+    @Command(name = "set-target-protocol", description = "Set or change the target consensus protocol of the specified tables. If a migration is in progress then the migration will be reversed with migrating ranges still migrating, unmigrated ranges marked as migrated, and migrating ranges will need migration. Be aware that if no migration was in progress for a table it will immediately cause the table to run on the target protocol because the ranges requiring migration are derived from the migrated ranges that don't exist.")
+    public static class SetTargetProtocol extends ConsensusMigrationAdmin
+    {
+        @Arguments(usage = "[<keyspace> <tables>...]", description = "The keyspace followed by one or many tables")
+        private List<String> schemaArgs = new ArrayList<>();
+
+        @Option(title = "target_protocol", name = {"-tp", "--target-protocol"}, description = "Use -tp to specify what consensus protocol should be migrated to", required=true)
+        private String targetProtocol = null;
+
+        @Option(title = "force_completion", name = {"-f", "--force-completion"}, description = "Nuclear option to force the migration state for all ranges in the table regardless")
```

Comment:
how about `Forces migration state for all ranges of the specified table regardless of whether migration completed successfully or not. Should only be used if table is empty or has had no writes since last repair.`. This does seem useful in some scenarios, so I don't think the docs should be too scary, but we should be explicit about when it's ok to use.

Reply:
Sure. That doesn't really communicate that this is correctness impacting.